src/objects/movers/PathPlayerMover.h

```diff
@@ -17,7 +19,7 @@ class PathPlayerMover {
         player(player), path(path) {
```

Comment:
@onthebluedanube one warning I use to get when I run my application is related to this line. It says 👇 

```c++
/Users/vladino/CLionProjects/mygame/src/objects/movers/PathPlayerMover.h:19:30: warning: field 'path' is uninitialized when used here [-Wuninitialized]
   19 |         player(player), path(path) {
      |                              ^
1 warning generated.
```

Would you rather define Path as std::unique_ptr so that we can defer initialization to constructor body?

Reply:
I would do something like this, instead of initializing in the body:

```suggestion
        player(player), path(player.GetPosition(), data_dir() /= std::filesystem::path("resources/images/route.png"), terrainSize) {
```

Yes, a reference always has to be bound to something, and the only way to ensure that is to initialize it in the member initializer list.